pkg/gui/controllers/helpers/branches_helper.go

```diff
@@ -97,6 +97,59 @@ func (self *BranchesHelper) ConfirmDeleteRemote(remoteName string, branchName st
 	return nil
 }
 
+func (self *BranchesHelper) ConfirmLocalAndRemoteDelete(branch *models.Branch) error {
+	if self.checkedOutByOtherWorktree(branch) {
+		return self.promptWorktreeBranchDelete(branch)
+	}
+
+	isMerged, err := self.c.Git().Branch.IsBranchMerged(branch, self.c.Model().MainBranches)
+	if err != nil {
+		return err
+	}
+
+	prompt := utils.ResolvePlaceholderString(
+		self.c.Tr.DeleteLocalAndRemoteBranchPrompt,
+		map[string]string{
+			"localBranchName":  branch.Name,
+			"remoteBranchName": branch.UpstreamBranch,
+			"remoteName":       branch.UpstreamRemote,
+		},
+	)
+
+	if !isMerged {
+		prompt += "\n\n" + utils.ResolvePlaceholderString(
+			self.c.Tr.ForceDeleteBranchMessage,
+			map[string]string{
+				"selectedBranchName": branch.Name,
+			},
+		)
+	}
```

Comment:
I decided to put up only a single confirmation ("are you sure you want to delete both branches"). If the local branch is not merged, I append the warning about this below in the same popup. I like this better than putting up a separate popup for this after the first one, but I can change it if you disagree.